src/build/env.py

```diff
@@ -246,20 +246,17 @@ def _find_executable_and_scripts(path):  # type: (str) -> Tuple[str, str, str]
         """
         config_vars = sysconfig.get_config_vars().copy()  # globally cached, copy before altering it
         config_vars['base'] = path
-        env_scripts = sysconfig.get_path('scripts', vars=config_vars)
-        if not env_scripts:
-            raise RuntimeError("Couldn't get environment scripts path")
-        exe = 'pypy3' if platform.python_implementation() == 'PyPy' else 'python'
-        if os.name == 'nt':
-            exe = '{}.exe'.format(exe)
-        executable = os.path.join(env_scripts, exe)
+        is_apple_python = 'osx_framework_library' in sysconfig.get_scheme_names()
+        paths = (
+            sysconfig.get_paths(scheme='posix_prefix', vars=config_vars)
+            if is_apple_python
+            else sysconfig.get_paths(vars=config_vars)
+        )
```

Comment:
Personally, I think the following is clearer:

```suggestion
        if 'osx_framework_library' in sysconfig.get_scheme_names():  # apple python
            paths = sysconfig.get_paths(scheme='posix_prefix', vars=config_vars)
        else:
            paths = sysconfig.get_paths(vars=config_vars)
```

Reply:
Maybe add a small comment there too explaining this.

Reply:
Sure, that works.